Actions/Deploy/Deploy.ps1

```diff
@@ -10,6 +10,85 @@ Param(
     [string] $deploymentEnvironmentsJson
 )
 
+function InstallOrUpgradeApps {
+    Param(
+        [hashtable] $bcAuthContext,
+        [string] $environment,
+        [string[]] $apps,
+        [string] $installMode
+    )
+
+    $schemaSyncMode = 'Add'
+    if ($installMode -eq 'ForceUpgrade') {
+        $schemaSyncMode = 'Force'
+        $installMode = 'upgrade'
+    }
+    $tempPath = Join-Path ([System.IO.Path]::GetTempPath()) ([GUID]::NewGuid().ToString())
+    New-Item -ItemType Directory -Path $tempPath | Out-Null
+    try {
+        Copy-AppFilesToFolder -appFiles $apps -folder $tempPath | Out-Null
+        $apps = @(Get-ChildItem -Path $tempPath -Filter *.app | ForEach-Object { $_.FullName })
+        $installedApps = Get-BcInstalledExtensions -bcAuthContext $bcAuthContext -environment $environment | Where-Object { $_.isInstalled }
+        $PTEsToInstall = @()
+        # Run through all apps and install or upgrade AppSource apps first (and collect PTEs)
+        foreach($app in $apps) {
+            # Get AppJson (works for full .app files, symbol files and also runtime packages)
+            $appJson = Get-AppJsonFromAppFile -appFile $app
+            $isPTE = ($appjson.idRanges.from -lt 100000 -and $appjson.idRanges.from -ge 50000)
+            $installedApp = $installedApps | Where-Object { $_.id -eq $appJson.id }
+            $needsUpgrade = $false
+            $needsInstall = $false
+            if ($installedApp) {
+                $newVersion = [version]::new($appJson.Version)
+                $installedVersion = [version]::new($installedApp.versionMajor, $installedApp.versionMinor, $installedApp.versionBuild, $installedApp.versionRevision)
+                if ($newVersion -gt $installedVersion) {
+                    $msg = "Dependency app $($appJson.name) is already installed in version $installedVersion, which is lower than $newVersion."
+                    if ($installMode -eq 'upgrade') {
+                        Write-Host "$msg Needs upgrade."
+                        $needsUpgrade = $true
+                    }
+                    else {
+                        Write-Host "::WARNING::$msg Set DependencyInstallMode to 'upgrade' or 'forceUpgrade' to upgrade dependencies."
```

Comment:
This warning might not be necessary. Even in _BCApps_ the dependencies are always "major.minor.0.0". 
I imagine bumping the dependency version is only needed if you want to uptake functionality from a newer version. 
In that sense, a dependency version is the "minimum required version", so doesn't need to be updated all the time.

Reply:
I will remove the github warning, but keep the output